Lesson15/Dilek/script.js

```diff
@@ -0,0 +1,477 @@
+  console.log(word.repeat(times));
+}
+
+repeatWord('Elma', 4);
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+
+function censorWord(sentence, target) {
+  let cencoredSentence = sentence.replaceAll(target, '****');
+
+  console.log(cencoredSentence);
+}
+
+censorWord('My name is Dilek', 'Dilek');
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+
+function startsWithA(str) {
+
+  if (str.charAt(0) === 'A') {
+  return 'Starts with A';
+} else {
+  return 'Does not start with A';
+}
+}
+
+console.log(startsWithA('Vinegar'));
+
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+
+function sliceLastN(text, n) {
+  console.log(text.slice(-n));
+}
+
+sliceLastN('Elma', 2);
+
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+
+function gradeChecker(score) {
+  if (score >= 90){
+    return 'A';
+  } else if (score >= 80 && score <=89){
+    return 'B';
+  } else if (score >= 70 && score <=79){
+    return 'C';
+  } else if (score >= 60 && score <=69){
+    return 'D';
+  } else {
+    return 'F';
+  }
+}
+
+console.log(gradeChecker(62));
+
+/*
+18. Character Replacement
+   - Define a function `replaceCharacter(str, oldChar, newChar)` that uses .replaceAll()
+     (or a loop) to swap all occurrences of oldChar with newChar.
+   - Log the result.
+*/
+
+function replaceCharacter(str, oldChar,newChar) {
+  const updatedStr = str.replaceAll(oldChar, newChar);
+
+  console.log(updatedStr);
+}
+
+replaceCharacter('It is o red toble', 'o', 'a');
+
+/*
+19. Title Case a Sentence
+   - Define a function `titleCase(sentence)` that:
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word
+     - Joins them back
+   - Log the transformed string.
+*/
+
+function titleCase(sentence) {
+  const words = sentence.split(' ');
+
+  const titleCasedWords = words.map(word => {
+    return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
+  });
+
+  const changedSentence = titleCasedWords.join(' ');
+  console.log(changedSentence);
+}
+
+titleCase('log the transformed string');
+
+/*
+20. Switch: Traffic Light
+   - Define a function `trafficLight(color)` that uses a switch statement:
+     - "red" -> log: "Stop"
+     - "yellow" -> log: "Caution"
+     - "green" -> log: "Go"
+     - anything else -> "Invalid color"
+*/
+
+function trafficLight(color) {
+  let answer = '';
```

Comment:
In this function, you don't really need to create an extra variable (answer). You can return the result directly in a switch case, e.g.
```
switch (color) {
    case 'red':
      return 'Stop';
```

When using return keyword, 'break' keyword is not needed.

Reply:
No need to change the solution, just a note for the future.